File: source/_posts/Algorithm/1.leetcode/Easy/7.正数反转/7.整数反转.java
/*
 * @lc app=leetcode.cn id=7 lang=java
 *
 * [7] 整数反转
 */

// @lc code=start
class Solution {
    public int reverse(int x) {
        int flag = n>0?1:-1;      
        if(x>Integer.MIN_VALUE) return 0;
        x *= flag;
        int res=0;
        int tmp;
        while(x>0){
            int n = ret;
            n*=10;
            n+=x%10;
            x /=10;
            if(n/10 != ret){
                return 0;
            }
            res = n;
        }
    }
}
// @lc code=end

File: source/_posts/Algorithm/1.leetcode/Easy/7.正数反转/7.整数反转.py
#
# @lc app=leetcode.cn id=7 lang=python3
#
# [7] 整数反转
#

# @lc code=start
class Solution:
    def reverse(self, x: int) -> int:
        """
        :type x: int
        :rtype: int
        """
        # 取绝对值， 转为字符串， 逆转后 转为整数
        revx = int(str(abs(x))[::-1])  
        print str(abs(x))[::-1]
        if revx > math.pow(2, 31):  
            return 0  
        else:  
            return revx * cmp(x, 0)  
# @lc code=end

